app/routes/contracted-hours.js

```diff
@@ -83,3 +94,13 @@ module.exports = function (router) {
     }
   }
 }
+
+const validRole = function (req) {
+  return authorisation.hasRole(req, roles.OFFENDER_MANAGER)
```

Comment:
You could update `authorisation.hasRole` to accept an array of roles and an error message. It would then, ensure the user is authenticated, check if the user has any of the roles specified and call `accessDenied` internally.

This would remove duplication and allow you to replace:
```
if (!authorisation.isUserAuthenticated(req)) {
  return res.redirect('/login')
}
if (!validRole(req) { return denieAccess(res) }
```
with:
```
authorisation.hasRole(req, [roles.OFFENDER_MANAGER])
```
and remove all of the `validRole` and `denieAccess` methods in the admin routes.